include/emp/web/Color.hpp

```diff
@@ -0,0 +1,459 @@
+        } else if (h > 1.0f) {
+            h -= 1.0f;
+        }
+
+        if (h * 6.0f < 1.0f) {
+            return m1 + (m2 - m1) * h * 6.0f;
+        }
+        if (h * 2.0f < 1.0f) {
+            return m2;
+        }
+        if (h * 3.0f < 2.0f) {
+            return m1 + (m2 - m1) * (2.0f / 3.0f - h) * 6.0f;
+        }
+        return m1;
+    }
+
+    emp::vector<std::string> split(const std::string& s, char delim) {
+        emp::vector<std::string> elems;
+        std::stringstream ss(s);
+        std::string item;
+        while (std::getline(ss, item, delim)) {
+            elems.push_back(item);
+        }
+        return elems;
+    }
+
+  }
+
+  Color::Color( const std::string& css_str ) {
+   std::string str = css_str;
+
+    // Remove all whitespace, not compliant, but should just be more accepting.
+    emp::remove_whitespace(str);
+
+    // Convert to lowercase.
+    emp::to_lower(str);
+
+    // Ensure sorted and find named color match with lower_bound
+    emp_assert(std::is_sorted(
+        std::begin(color_impl::namedColors), std::end(color_impl::namedColors), 
+        [](const auto& left, const auto& right){
+            return left.name < right.name;
+        }));
+    
+    auto namedColor = std::lower_bound(
+        std::begin(color_impl::namedColors), std::end(color_impl::namedColors), str,
+        [](const auto& left, const auto& right){
+            return left.name < right;
+        });
+
+    if (namedColor->name == str){
+        *this = namedColor->color;
+        (*this).css_str = emp::ColorRGB(r, g, b, a);
+        return;
+    }
+
+    *this = ParseColor(str);
+
+  }
+  
+  // Handle all color detection and parsing from the string
+  Color Color::ParseColor(const std::string& str) {
+      // #abc and #abc123 syntax.
+    if (DetectSyntaxABC(str)) {
+        return ParseABC(str);
+    }
+
+    size_t open_paren = str.find_first_of('(');
+    size_t close_paren = str.find_first_of(')');
+    if (open_paren != std::string::npos && close_paren + 1 == str.length()) {
+        const std::string fname = str.substr(0, open_paren);
+        const std::vector<std::string> color_tokens = color_impl::split(str.substr(open_paren + 1, 
+            close_paren - (open_paren + 1)), ',');
+
+        float alpha = 1.0f;
+        if (fname == "rgba" || fname == "rgb") {
+            return ParseRGB(str, fname, color_tokens, alpha);
+
+        } else if (fname == "hsla" || fname == "hsl") {
+            return ParseHSL(str, fname, color_tokens, alpha);
+        }
+    }
+    emp_always_assert( false, str);
+    __builtin_unreachable();
+  }
+
+  Color Color::ParseABC(const std::string& str){
+      if (str.length() == 4) {
+            const int64_t iv = color_impl::parseInt(str.substr(1), 16);  // TODO(deanm): Stricter parsing.
+            if (!(iv >= 0 && iv <= 0xfff)) {
+                emp_assert( false, str );
+                __builtin_unreachable();
+            } else {
+                Color* col = new Color(
```

Comment:
this will leak memory. That color that we new will never get deleted. Just return Color(
...
...
);

instead

Reply:
looks like this issue is happening on several other lines